### PR TITLE
FM2-248:-Update MedicationFhirResource Provider delete method.

### PR DESCRIPTION
**Pull request by Akayeshmantha**
Update MedicationFhirResource Provider delete method.

## Issue I worked on
see https://issues.openmrs.org/browse/FM2-248

## Checklist: I completed these to help reviewers :)
- [x ] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [ x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x ] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [ x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/fhir2/providers/r3/MedicationFhirResourceProvider.java [hunk 1]
```diff
@@ -22,8 +22,8 @@
 import org.hl7.fhir.convertors.conv30_40.Medication30_40;
 import org.hl7.fhir.dstu3.model.IdType;
 import org.hl7.fhir.dstu3.model.Medication;
-import org.hl7.fhir.dstu3.model.OperationOutcome;
 import org.hl7.fhir.instance.model.api.IBaseResource;
+import org.hl7.fhir.r4.model.OperationOutcome;
```
[the remark this answers is not in the document]
Comment (Akayeshmantha) [reply api/src/main/java/org/openmrs/module/fhir2/providers/r3/MedicationFhirResourceProvider.java:26]: @corneliouzbett  the util method depends on the import org.hl7.fhir.r4.model.OperationOutcome;
so it returns r4 type not dtsu3 type

[the remark this answers is not in the document]
Comment (Akayeshmantha) [reply api/src/main/java/org/openmrs/module/fhir2/providers/r3/MedicationFhirResourceProvider.java]: done

**Comment by Akayeshmantha** [comment]
@corneliouzbett  added changes.

**Comment by corneliouzbett** [comment]
Thanks @Akayeshmantha for this!

**Pull request merged**